erts/preloaded/src/erlang.erl

```diff
@@ -10456,11 +10456,9 @@ the new process, atomically. Otherwise works like `spawn/3`.
 """.
 -doc #{ category => processes }.
 -spec spawn_link(Fun) -> pid() when
-      Fun :: function().
-spawn_link(F) when erlang:is_function(F) ->
+      Fun :: fun(() -> term()).
+spawn_link(F) when erlang:is_function(F, 0) ->
     erlang:spawn_link(erlang, apply, [F, []]);
-spawn_link({M,F}=MF) when erlang:is_atom(M), erlang:is_atom(F) ->
-    erlang:spawn_link(erlang, apply, [MF, []]);
```

Comment:
The `{M, F}` form is a holdover from when funs were tuples. This 2-tuple form was used to represent exported functions (`fun M:F/A`), and worked all the way until ~R15 in 2011.